fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/LoanDelinquencyDomainServiceImpl.java

```diff
@@ -66,10 +69,11 @@ public CollectionData getOverdueCollectionData(final Loan loan) {
 
                         amountAvailable = installment.getTotalPaid(loanCurrency).getAmount();
 
-                        for (LoanTransactionToRepaymentScheduleMapping mappingInstallment : installment
-                                .getLoanTransactionToRepaymentScheduleMappings()) {
-                            final LoanTransaction loanTransaction = mappingInstallment.getLoanTransaction();
-                            if (loanTransaction.isChargeback()) {
+                        boolean isLatestInstallment = installment.equals(latestInstallment);
```

Comment:
Be careful with this. The LoanRepaymentScheduleInstallment doesn't have the equals/hashCode overriden. The often given advice is to do equality checks based on identity and override the equals/hashCode accordingly.
Can't we check the ID here instead of the full object?